streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractReadOnlyDecorator.java

```diff
@@ -45,12 +46,19 @@ public void flush() {
         throw new UnsupportedOperationException(ERROR_MESSAGE);
     }
 
+    @Deprecated
     @Override
     public void init(final ProcessorContext context,
                      final StateStore root) {
         throw new UnsupportedOperationException(ERROR_MESSAGE);
     }
 
+    @Override
+    public void init(final StateStoreContext context,
+                     final StateStore root) {
+        throw new UnsupportedOperationException(ERROR_MESSAGE);
```

Comment:
There are going to be a lot of duplicated init methods. It's not great, but hopefully we can drop the old API before too long.